go/master/service.go

```diff
@@ -77,11 +77,13 @@ type taskEntry struct {
 	NumFailure int
 }
 
-type taskQueues struct {
-	Todo    []taskEntry
-	Pending map[int]taskEntry // map from task ID to task entry
-	Done    []taskEntry
-	Failed  []taskEntry
+type masterState struct {
+	Todo     []taskEntry
+	Pending  map[int]taskEntry // map from task ID to task entry
+	Done     []taskEntry
+	Failed   []taskEntry
+	CurPass  int
+	JobTasks []taskEntry
```

Comment:
You are right, just realized that we don't need `JobTasks` any more, replaced
```go
s.state.TODO = s.state.JobTasks
```
with:
```go
s.state.TODO = append(s.state.Done, s.state.Failed...)
```